kim-dijkstras-oop/src/js/Graph.js

```diff
@@ -0,0 +1,90 @@
+import Node from './Node'
+'use strict'
+
+export default class Graph {
+  constructor(nodes) {
+    if (nodes) {
+      this._nodes = nodes;
+    } else {
+      this._nodes = [];
+    }
+  }
+
+  getNrOfNodes() {
+    return this._nodes.length;
+  }
+
+  findNode(name) {
+    var node = null;
+    if (this._nodes[0].getName() === name) {
+      return this._nodes[0];
+    }
+    for (var i = 1; i < this._nodes.length && this._nodes[i - 1].getName() !== name; i++) {
+      node = this._nodes[i];
+    }
+    return node;
+  }
+
+  addNode(name) {
+    var node = new Node(name);
+    this._nodes.push(node);
+
+    return node;
+  }
+
+  dijkstras(startNode, endNode) {
+    var dist = [];
+    var path = [];
+    var unvisited = this._nodes.slice();
+    var visited = [];
+    var current;
+    var currentIndex;
+    var currentNeighbors;
+    var resultingPath = {
+      'path': [],
+      'dist': 0
+    }
+
+    if (startNode === undefined || endNode === undefined) {
+      return resultingPath;
+    } else if (startNode === endNode) {
+      return resultingPath;
+    }
+
+    for (var i = 0; i < this._nodes.length; i++) {
+      dist[this._nodes[i].getName()] = Infinity;
+    }
+    path[startNode.getName()] = [];
+    dist[startNode.getName()] = 0;
+
+    while (unvisited.length > 0 && unvisited.includes(endNode)) {
+      current = unvisited[0];
+      currentIndex = 0;
```

Comment:
This reset of variables is unnecessary, since the forloop get the smallest value afterwards and puts it in the variable